pkg/testing/fixture.go

```diff
@@ -1044,6 +1043,33 @@ func (f *Fixture) DumpProcesses(suffix string) {
 	}
 }
 
+// KeepFileByMoving moves file to 'build/diagnostics' which contents are
+// available on CI if the test fails or on the agent's 'build/diagnostics'
+// if the test is run locally.
+// 'prefix is added to the file name when moving
+func (f *Fixture) KeepFileByMoving(file, prefix string) {
```

Comment:
Nit: a clearer name would help, maybe even splitting the 2 things that this function is doing in multiple functions
```suggestion
func (f *Fixture)BuildArtifactsDir() {...}
func (f *Fixture)MoveFile(src, dst string) {...}
func (f *Fixture)MoveToArtifacts(file string) {
	dstFileDir := f.BuildArtifactsDir()
	dstFileName := f.FileNamePrefix() + filepath.Base(file)
	f.MoveFile(file, filepath.Join(dstFileDir, dstFileName))
}
```